src/excmds.ts

```diff
@@ -4389,289 +4383,182 @@ export async function hint(option?: string, selectors?: string, ...rest: string[
-        case "-t": // Open in foreground
-            selectHints = hinting.pipe(
-                withSelectors ? [selectors, ...rest].join(" ") : DOM.HINTTAGS_selectors,
-                async link => {
-                    link.focus()
-                    if (link.href) {
-                        hintTabOpen(link.href, option === "-t").catch(() => DOM.simulateClick(link))
-                    } else {
-                        DOM.simulateClick(link)
-                    }
-                    return link
-                },
-                rapid,
-                jshints,
-            )
-            break
 
-        case "-y":
-            // Yank link
-            selectHints = hinting.pipe(
-                'a[href]:not([href="#"])',
-                elem => {
-                    // /!\ Warning: This is racy! This can easily be fixed by adding an await but do we want this? yank can be pretty slow, especially with yankto=selection
-                    yank(elem.href)
-                    return elem
-                },
-                rapid,
-                jshints,
-            )
-            break
+    return new Promise((resolve, reject) => {
+        const hintables = config.hintables()
 
-        case "-p":
-            // Yank text content
-            selectHints = hinting.pipe_elements(
-                DOM.elementsWithText(),
-                elem => {
-                    // /!\ Warning: This is racy! This can easily be fixed by adding an await but do we want this? yank can be pretty slow, especially with yankto=selection
-                    yank(elem.textContent)
-                    return elem
-                },
-                rapid,
-            )
-            break
+        // If the user specified a callback, eval it, else use the default
+        // action which performs the action matching the open mode
+        const action = config.callback
+            ? eval(config.callback)
+            : (elem: any) => {
+                  if (config.pipeAttribute !== null) {
+                      // We have an attribute to pipe
+                      return elem[config.pipeAttribute]
+                  }
 
-        case "-h":
-            // Highlight element
-            selectHints = hinting.pipe_elements(
-                DOM.elementsWithText(),
-                elem => {
-                    const r = document.createRange()
-                    r.setStart(elem, 0)
-                    r.setEnd(elem, 1)
-                    const s = document.getSelection()
-                    s.addRange(r)
-                    return elem
-                },
-                rapid,
-            )
-            break
+                  if (config.excmd) {
+                      // We have an excmd to run. By spec, we append the element's href
+                      if (elem.href) {
+                          // /!\ RACY RACY RACY!
+                          run_exstr(config.excmd + " " + elem.href)
```

Comment:
I think it should be possible to do this a bit more safely:
```suggestion
                          CTSELF["config.excmd"](elem.href)
```

Reply:
I don't know about this, I just took the code for the various callbacks as-is and added it in the corresponding spot while refactoring. With the scope of this PR being already large enough I don't know changing even more things in the same batch?

Reply:
Yeah, you're right, it's kind of out of scope.